### PR TITLE
feat(types): add types for `page.$$eval`

### PR DESCRIPTION
**Pull request by jackfranklin**

### CODE REVIEW

**Comment by jackfranklin** [comment]
@hanselfmu PTAL, thanks :)

File: src/common/Page.ts [hunk 1]
```diff
@@ -795,31 +794,70 @@ export class Page extends EventEmitter {
...
+   * // or if you want to be more explicit, provide it as the generic type.
+   * const allInputValues = await page.$$eval<string[]>(
+   *  'input', (elements: HTMLInputElement[]) => elements.map(e => e.textContent)
+   * );
+   * ```
+   *
+   * @param selector the
    * {@link https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors | selector}
    * to query for
-   * @param pageFunction - the function to be evaluated in the page context.
```
Comment (hanselfmu) [inline src/common/Page.ts:811]: Will API Extractor report a warning here if we remove the `-` after `pageFunction`?

**Review by hanselfmu — APPROVED** [review]
LGTM % the `@param -` thing, but it shouldn't block this from landing.

File: src/common/Page.ts (hunk 1 above)
Comment (jackfranklin) [reply src/common/Page.ts:811]: yes, it needs there to be a `-` between the param name and the description

**Pull request merged**